recipe/build-mpi.sh

```diff
@@ -59,11 +68,7 @@ fi
 ./configure --prefix=$PREFIX \
             --disable-dependency-tracking \
             --enable-mpi-fortran \
-            --disable-wrapper-rpath \
-            --disable-wrapper-runpath \
```

Comment:
looks like the rpath in wrapper_ldflags is required on mac at least, because it's now failing with:

```
mpicc helloworld.c -o helloworld_c
+ mpiexec -n 4 ./helloworld_c
dyld[61747]: Library not loaded: '@rpath/libmpi.40.dylib'
  Referenced from: '/Users/runner/miniforge3/conda-bld/openmpi-mpi_1722332831711/test_tmp/tests/helloworld_c'
  Reason: tried: '/usr/local/lib/libmpi.40.dylib' (no such file), '/usr/lib/libmpi.40.dylib' (no such file)
dyld[61748]: Library not loaded: '@rpath/libmpi.40.dylib'
  Referenced from: '/Users/runner/miniforge3/conda-bld/openmpi-mpi_1722332831711/test_tmp/tests/helloworld_c'
  Reason: tried: '/usr/local/lib/libmpi.40.dylib' (no such file), '/usr/lib/libmpi.40.dylib' (no such file)
dyld[61750]: Library not loaded: '@rpath/libmpi.40.dylib'
  Referenced from: '/Users/runner/miniforge3/conda-bld/openmpi-mpi_1722332831711/test_tmp/tests/helloworld_c'
  Reason: tried: '/usr/local/lib/libmpi.40.dylib' (no such file), '/usr/lib/libmpi.40.dylib' (no such file)
dyld[61751]: Library not loaded: '@rpath/libmpi.40.dylib'
  Referenced from: '/Users/runner/miniforge3/conda-bld/openmpi-mpi_1722332831711/test_tmp/tests/helloworld_c'
  Reason: tried: '/usr/local/lib/libmpi.40.dylib' (no such file), '/usr/lib/libmpi.40.dylib' (no such file)
```

suggesting that wrapper-rpath is wrong on mac, at least. Maybe this is why we do it ourselves. Looking at the showme output, the wrappers don't set rpath at all on mac for some reason.

`runpath` is also excluded with a comment related to `enable-new-dtags`. I don't know the consequences of that one, so I don't know if it still needs to be omitted. I'd suggest leaving these as they are unless it's known that the fixes are no longer required.

Reply:
I fixed the macOS part. Indeed, in macOS we do need rpath, because conda-forge change the dylib name to `@rpath/libmpi.*.dylib`, therefore an explicit rpath is needed.

Regarding Linux, `--enable-new-dtags` is what we need if we want the LD_LIBRARY_PATH mechanism to allow for changing the MPI library at runtime. If you look at the logs for `mpicc -show`, you will see that  `--enable-new-dtags` is there, Open MPI is checking for it and adding it automatically. BTW, that's also the case with MPICH 

Are we good or did I misunderstand your comments/concerns? 
Could you point me to the exact comment that worried you?

Reply:
It might be fine. My only concern is that disabling dtags was done explicitly [on purpose](https://github.com/conda-forge/openmpi-feedstock/pull/35/commits/91277bcbd58f3c0d17e6cc75cb7299bb383c38c1), butI no longer remember what problem the new dtags caused, so I can't speak to whether it's still relevant or not or how to check.

But if you're confident and/or happy to field problems if this causes any, I'm okay, too. I like the simplification.

Reply:
> Don't you remember why you made such choice?

I assume this is a joke, since it is hilarious to think that I might remember every fiddly little decision from 6 years ago.

Reply:
Well, to be honest, it was not a joke, but I did not realize that the change was done so long ago!